lib/services/oidc.go

```diff
@@ -90,6 +93,23 @@ func OIDCClaimsToTraits(claims jose.Claims) map[string][]string {
 	return traits
 }
 
+// GetRedirectURL gets a redirect URL for the given connector. If the connector
+// has a redirect URL which matches the host of the given Proxy address, then
+// that one will be returned. Otherwise, the first URL in the list will be returned.
+func GetRedirectURL(conn types.OIDCConnector, proxyAddr string) string {
+	if len(conn.GetRedirectURLs()) == 0 {
+		return ""
+	}
+
+	for _, redirectURL := range conn.GetRedirectURLs() {
+		if strings.Contains(redirectURL, proxyAddr) {
```

Comment:
Wonder if `string.Contains` check is sufficient for this case.  

For example if proxy a client have 2 proxy with publicAddrs set to `teleport.example.com` and `dev.teleport.example.com` the if will be evaulted to true for  `redirectURL(dev.teleport.example.com/v2/web/oidc/..)` and `proxyAddr(teleport.example.com)`

Reply:
Agreed, at first I wasn't sure what to do here, so I took the naive approach.

I'll replace this with a "best effort matching" approach. First I'll match host, then port and schema. If there is at least one with a matching host, then the best overall match will be chosen.

Any other ideas would be welcome